src/mgmt/srv_query.c

```diff
@@ -495,14 +484,10 @@ ds_mgmt_dev_state_query(uuid_t dev_uuid, Ctl__DevStateResp *resp)
 		return rc;
 	}
 
-	D_ALLOC(resp->dev_state, buflen);
-	if (resp->dev_state == NULL) {
-		D_ERROR("Failed to allocate device state");
-		rc = -DER_NOMEM;
-		goto out;
-	}
-	strncpy(resp->dev_state,
-		smd_dev_stat2str(dev_info->sdi_state), buflen);
+	if (dev_info->sdi_state == SMD_DEV_NORMAL)
+		resp->dev_state = BIO_DEV_NORMAL;
+	else
+		resp->dev_state = BIO_DEV_FAULTY;
```

Comment:
Is this to query the state of a specific device? If so, why don't we return the exact state (like what we did in list dev operation)?

I think control plane may display a verbose state for such specific query, like "in use: yes, faulty: yes, plugged: yes".  @sydneyvanda what's your opinion?

Reply:
I'm struggling to understand how we do that when we have different states SMD BIO NVME, how should they be mapped?

Reply:
in this context should we just be translating bdi_flags to BIO_ ones?

Reply:
I think the BIO_DEV_XXX could be removed, it looks to me just a subset of NVME_DEV_FL_XXX, and we'd pass NVME_DEV_FL_XXX directly to control plane.

I don't think we'd expose SMD_DEV_XXX to control plane. (It should be used by BIO internally)

Here on query a specific device, I think the device state returned to control plane should be consistent with the device state returned by 'device list', so here we may need to use the similar way used in 'device list' to get the device state. @sydneyvanda , what do you think?

Reply:
sounds good to me

Reply:
Looks this ds_mgmt_dev_state_query() can only query in-use device (the device stored in SMD), so I think we can just return "FL_INUSE | FL_PLUGGED" when SMD sate is NORMAL, and return "FL_INUSE | FL_PLUGGED | FL_FAULTY" when SMD state is FAULTY.

I think this per device query could be improved in the future, all devices listed by "device list" should be query-able, and the device state returned by query should be consistent with the state returned by "device list".